### ISSUE TITLE
Work with base path

### ISSUE THREAD

**Issue by vincentjames501** [danieltdt/node-json-rest-client#3]
Maybe this is just a documentation issue, but right now I have to add my 'base path' to each of my resource definitions:

```
var rest = require('json-rest-client')({
        url: 'https://myapi.com/api/v1'
    }, {
        headers: {
            'X-Client-Version': 'NodejsClient 0.0.1',
            'Authorization': 'Bearer ' + config.token
        }
    });

return {
        users: rest.resource('/api/v1/users') // Have to include path here otherwise it doesn't work
    };
```

It would be nice to have just the path appended to the base path.

**Comment by danieltdt** [comment]
Actually, `restify` [has an issue](https://github.com/mcavage/node-restify/pull/21) about this but people don't seem to care about that xD
However, I can add a new option for `requestOptions` that prefixes all routes.

```js
var rest = require('json-rest-client')({url: 'https://myapi.com'}, {prefix: '/api/v1'});
var api = {users: rest.resources('/users')};

api.users(2).show();
// GET /api/v1/users/2
```

**Comment by vincentjames501** [comment]
That would be great!

**Comment by danieltdt** [comment]
Released at v1.2.0
